packages/flutter/lib/src/material/button_style.dart

```diff
@@ -435,3 +435,33 @@ class _LerpShapes implements MaterialStateProperty<OutlinedBorder?> {
     return ShapeBorder.lerp(resolvedA, resolvedB, t) as OutlinedBorder?;
   }
 }
+
+/// How the icon in any `button.icon` should be placed, where `button` could be:
+///
+/// * [TextButton]
+/// * [ElevatedButton]
+/// * [OutlinedButton]
+///
+/// In buttons that have an icon, we generally tend to use a `Row` widget
+/// to be able to put multiple widgets in a horizontal way, aka `Row`.
+///
+/// Hover over enum values to be able to see more deailed description
+/// on what do they do.
+enum ButtonIconAlignment {
+  /// Places the icon to the `start` of the button which means the
+  /// first child in the `Row`.
+  ///
+  /// The `Row` will be like this if you choose to use `start` which is
+  /// the default:
+  ///
+  /// `[YOUR_ICON, SIZEDBOX, LABEL]`
```

Comment:
This is nice and detailed but I think we'd be better off avoiding saying exactly how it's implemented. Another type of button could use a different implementation but this enum could still apply.

Also we should say what "start" means.

Reply:
thanks, I have updated the docs to state what `start` means by talking about the horizontal axis and where relative to the button it will be.